packages/@aws-cdk/aws-appmesh/lib/mesh.ts

```diff
@@ -0,0 +1,368 @@
+   *
+   * @abstract
+   * @type {string}
+   * @memberof MeshBase
+   */
+  public abstract readonly meshUid: string;
+
+  /**
+   * Adds a VirtualRouter to the Mesh with the given id and props
+   *
+   * @param {string} id
+   * @param {VirtualRouterBaseProps} props
+   * @returns {VirtualRouter}
+   * @memberof MeshBase
+   */
+  public addVirtualRouter(id: string, props: VirtualRouterBaseProps): VirtualRouter {
+    return new VirtualRouter(this, id, {
+      meshName: this.meshName,
+      virtualRouterName: props.virtualRouterName,
+      portMappings: props.portMappings,
+    });
+  }
+
+  /**
+   * Adds a VirtualService with the given id
+   *
+   * @param {string} id
+   * @param {VirtualServiceBaseProps} props
+   * @returns {VirtualService}
+   * @memberof MeshBase
+   */
+  public addVirtualService(id: string, props: VirtualServiceBaseProps): VirtualService {
+    return new VirtualService(this, id, {
+      ...props,
+      meshName: this.meshName,
+    });
+  }
+
+  /**
+   * Adds a VirtualNode to the Mesh
+   *
+   * @param {string} id
+   * @param {AddVirtualNodeProps} props
+   * @returns {VirtualNode}
+   * @memberof MeshBase
+   */
+  public addVirtualNode(id: string, props: AddVirtualNodeProps): VirtualNode {
+    return new VirtualNode(this, id, {
+      ...props,
+      meshName: this.meshName,
+      namespaceName: props.namespaceName,
+    });
+  }
+
+  /**
+   * Exports the Mesh properties to re-use in other stacks
+   */
+  public abstract export(): MeshImportProps;
+}
+
+/**
+ * The set of properties used when creating a Mesh
+ */
+export interface MeshProps {
+  /**
+   * The name of the Mesh being created
+   */
+  readonly meshName?: string;
+  /**
+   * The spec to be applied to the mesh, at this point, only egressFilter is available
+   */
+  readonly meshSpec?: MeshSpec;
+}
+
+/**
+ * Creates a new AppMesh mesh
+ *
+ * @see https://docs.aws.amazon.com/app-mesh/latest/userguide/meshes.html
+ */
+export class Mesh extends MeshBase {
+  /**
+   * A static method to import a mesh an make it re-usable accross stacks
+   *
+   * @static
+   * @param {cdk.Construct} scope
+   * @param {string} id
+   * @param {MeshImportProps} props
+   * @returns {IMesh}
+   * @memberof Mesh
+   */
+  public static import(scope: cdk.Construct, id: string, props: MeshImportProps): IMesh {
+    return new ImportedMesh(scope, id, props);
+  }
+
+  /**
+   * The name of the AppMesh mesh
+   *
+   * @type {string}
+   * @memberof Mesh
+   */
+  public readonly meshName: string;
+  /**
+   * The Amazon Resource Name (ARN) of the AppMesh mesh
+   *
+   * @type {string}
+   * @memberof Mesh
+   */
+  public readonly meshArn?: string;
+
+  /**
+   * The unique identifier for the mesh
+   *
+   * @type {string}
+   * @memberof Mesh
+   */
+  public readonly meshUid: string;
+
+  constructor(scope: cdk.Construct, id: string, props?: MeshProps) {
+    super(scope, id);
+
+    this.node.apply(new cdk.Tag(NAME_TAG, this.node.path));
```

Comment:
Is this necessary?

Reply:
necessary, no, I thought it might be useful to add a default name. I'm not married to it at all.